gazebo/rendering/Distortion.cc

```diff
@@ -186,7 +190,31 @@ void Distortion::SetCamera(CameraPtr _camera)
           idxV < this->dataPtr->distortionTexHeight)
       {
         unsigned int mapIdx = idxV * this->dataPtr->distortionTexWidth + idxU;
-        this->dataPtr->distortionMap[mapIdx] = uv;
+
+        // if the destination pixel has been mapped previously,
+        // need to avoid the distorted image "overlapping" itself
+        ignition::math::Vector2d n1; // a bit of a hack since vector2d overloads operator= but not operator!=
+        n1 = -1;
+        if (this->dataPtr->distortionMap[mapIdx] != n1)
```

Comment:
Probably a better way to do this; need to check if the value in this Vector2d object is equivalent to what it was set to initially (-1)